src/modules/menu/components/advancedTools/ImportFile/utils.js

```diff
@@ -39,31 +41,57 @@ export function isGpx(fileContent) {
  * @param {OBject} store Vuex store
  * @param {ArrayBuffer} content Content of the file
  * @param {string} source Source of the file (either URL or file path)
+ * @param {File | null} [originalFile]
  * @returns {ExternalLayer} External layer object
  */
-export async function handleFileContent(store, content, source) {
+export async function handleFileContent(store, content, source, originalFile = null) {
     let layer = null
-    let textContent
+    let parsedContent
     let linkFiles
     if (isZipContent(content)) {
         log.debug(`File content is a zipfile, assume it is a KMZ archive`)
         const kmz = await unzipKmz(content, source)
-        textContent = kmz.kml
+        parsedContent = kmz.kml
         linkFiles = kmz.files
+    } else if (source.endsWith('.tif')) {
+        log.debug(`File content might be a COGTIFF, attempting a parse as such`)
+        try {
+            parsedContent = await fromArrayBuffer(content)
+        } catch (err) {
+            log.debug('parsing as COGTIFF failed, moving on with other formats', err)
+        }
     } else {
         // If it is not a zip file then we assume is a text file and decode it for further handling
-        textContent = new TextDecoder('utf-8').decode(content)
+        parsedContent = new TextDecoder('utf-8').decode(content)
     }
-    if (isKml(textContent)) {
+    if (parsedContent instanceof GeoTIFF) {
+        layer = new GeoTIFFLayer({
+            fileSource: source,
+            visible: true,
+            opacity: 1.0,
+            // For local files : OpenLayers needs a Blob, and not an already parsed GeoTIFF instance
+            data: originalFile,
```

Comment:
Here the original file will be null if imported using the import tool file if I'm correct ?

Reply:
that's correct, it will be available through the `fileSource` prop